test/codegen/benchmark_delete_translator_test.cpp

```diff
@@ -0,0 +1,474 @@
+
+  timer.Start();
+  while (delete_executor->Execute()) {}
+
+  txn_manager.CommitTransaction(txn);
+
+  timer.Stop();
+  LOG_INFO("executor empty table finished in: %.2f ms\n", timer.GetDuration());
+  EXPECT_EQ(0, getCurrentTableSize());
+
+  reloadTable();
+}
+
+
+TEST_F(DeleteTranslatorTest, DeleteWithSimplePredicate) {
+  //
+  // DELETE FROM table where a == 10000; only one record shall be deleted
+  //
+
+  EXPECT_EQ(NumRowsInTestTable(), getCurrentTableSize());
+
+
+  // Setup the predicate
+  auto* a_col_exp =
+      new expression::TupleValueExpression(type::Type::TypeId::INTEGER, 0, 0);
+  auto* const_40_exp = CodegenTestUtils::ConstIntExpression(40);
+  auto* a_equal_40 = new expression::ComparisonExpression(
+      ExpressionType::COMPARE_EQUAL, a_col_exp, const_40_exp);
+
+  std::unique_ptr<planner::DeletePlan> delete_plan{
+      new planner::DeletePlan(&GetTestTable(TestTableId()), nullptr)};
+  std::unique_ptr<planner::AbstractPlan> scan{
+      new planner::SeqScanPlan(&GetTestTable(TestTableId()), a_equal_40, {0, 1, 2})};
+  delete_plan->AddChild(std::move(scan));
+
+  // Do binding
+  planner::BindingContext context;
+  delete_plan->PerformBinding(context);
+
+  // We collect the results of the query into an in-memory buffer
+  codegen::BufferingConsumer buffer{{0, 1, 2}, context};
+
+  Timer<std::ratio<1, 1000>> timer;
+
+  LOG_INFO("starting benchmarking empty table");
+  timer.Start();
+
+  // COMPILE and execute
+  CompileAndExecute(*delete_plan, buffer, reinterpret_cast<char*>(buffer.GetState()));
+
+  timer.Stop();
+  LOG_INFO("compiled empty table finished in: %.2f ms\n", timer.GetDuration());
+
+  EXPECT_EQ(NUM_OF_TEST_ROWS - 1, getCurrentTableSize());
+  reloadTable();
+
+  // =========================
+  //  Benchmark Executor mode
+  // =========================
+  auto &txn_manager =
+      concurrency::TransactionManagerFactory::GetInstance();
+
+  concurrency::Transaction *txn = txn_manager.BeginTransaction();
+
+  std::unique_ptr<executor::ExecutorContext> executor_context(
+      new executor::ExecutorContext(txn)
+  );
+
+  std::unique_ptr<executor::DeleteExecutor> delete_executor =
+      std::make_unique<executor::DeleteExecutor>(
+          delete_plan.get(), executor_context.get()
+      );
+
+  std::unique_ptr<executor::SeqScanExecutor> scan_executor =
+      std::make_unique<executor::SeqScanExecutor>(
+          delete_plan->GetChild(0), executor_context.get()
+      );
+
+  delete_executor->AddChild(scan_executor.get());
+
+  EXPECT_TRUE(delete_executor->Init());
+
+  timer.Reset();
+  timer.Start();
+  while (delete_executor->Execute()) {}
+  txn_manager.CommitTransaction(txn);
+
+  timer.Stop();
+  LOG_INFO("executor empty table finished in: %.2f ms\n", timer.GetDuration());
+  EXPECT_EQ(99999, getCurrentTableSize());
+
+  reloadTable();
+}
+
+TEST_F(DeleteTranslatorTest, DeleteHalfRecords) {
+  //
+  // DELETE FROM table where a % 20 == 0; half records will be deleted
```

Comment:
To vary the selectivity, maybe you can use a >= (?) so the predicate tree is simpler (though LLVM bakes the predicates in anyway)